.circleci/config.yml

```diff
@@ -151,33 +112,77 @@ jobs:
       - run:
           name: execute sonar-scanner
           command: >
-            sonar-scanner \
-              -Dsonar.login="${SONAR_TOKEN}" \
-              -Dsonar.projectVersion="${CIRCLE_BUILD_NUM}" \
-              -Dsonar.branch="${CIRCLE_BRANCH}"
+            if [ ! -z "$SONAR_TOKEN" ] && \
+              [ ! -z "$CIRCLE_BUILD_NUM" ] && \
+              [ ! -z "$CIRCLE_BRANCH" ]; then
+              sonar-scanner \
+                -Dsonar.login="${SONAR_TOKEN}" \
+                -Dsonar.projectVersion="${CIRCLE_BUILD_NUM}" \
+                -Dsonar.branch="${CIRCLE_BRANCH}"
+            else
+              echo "required env vars not found"
+            fi
       - save_cache:
           key: sonar-{{ epoch }}
           paths:
             - ~/.sonar
 
 
-  # dockerize:
-  #   <<: *defaults
-  #   machine: true
-  #   steps:
-  #     - attach_workspace:
-  #         at: /opt/iroha/build
-  #     - run: TODO
-  #         name: todo
-  #         command: echo "I am happy albatross"
-
+  macos-build:
+    #docker:
+    #  - image: postgres:9.5
+    #  - image: redis:3.2.8
+    macos:
+      xcode: 9.0.0
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - brew-
+          paths:
+            - Library/Caches/Homebrew
+      - run:
+          name: install dev dependencies
+          command: brew install cmake boost postgres grpc autoconf automake libtool ccache
+      - save_cache:
+          key: brew-{{ arch }}-{{ epoch }}
+          paths:
+            - Library/Caches/Homebrew
+      - restore_cache:
+          keys:
+            - build-cache-{{ .Branch }}-{{ arch }}
+          paths:
+            - ~/.ccache
+      - restore_cache:
+          keys:
+            - external-{{ arch }}-{{ .Branch }}
+            - external-{{ arch }}
+          paths:
+            - external
+      - run: ccache --show-stats
+      - run:
+          name: build
+          command: |
+            cmake -H. -Bbuild
```

Comment:
why not `cmake -H$IROHA_HOME -B$IROHA_BUILD`?

Reply:
there are possibly no $IROHA_HOME or $IROHA_BUILD